dap-mode.el

```diff
@@ -683,14 +693,37 @@ thread exection but the server will log message."
   (run-hook-with-args 'dap-terminated-hook debug-session)
   (dap--refresh-breakpoints))
 
+(defun dap--output-buffer-format-with-category (category output)
+  "Formats a string suitable for printing to the output buffer using CATEGORY and OUTPUT."
+  (let ((message (format "%s: %s" category output)))
+    (if (string= (substring message -1) "\n")
+        message
+      (concat message "\n"))))
+
+(defun dap--output-buffer-format (output-body)
+  "Formats a string suitable for printing to the output buffer using an OUTPUT-BODY."
+  (if dap-label-output-buffer-category
+      (dap--output-buffer-format-with-category (gethash "category" output-body) (gethash "output" output-body))
+    (gethash "output" output-body)))
+
+(defun dap--print-to-output-buffer (debug-session str)
+  "Insert content from STR into the output buffer associated with DEBUG-SESSION."
+  (with-current-buffer (dap--debug-session-output-buffer debug-session)
+    (let ((win (get-buffer-window (current-buffer))))
```

Comment:
I think that following the output might be achieved with https://stackoverflow.com/questions/12663061/emacs-auto-scrolling-log-buffer (e. g. when the user wants to follow the output). The problem with this implementation is that if you are in the debug console and you want to copy few lines at the beginning of the buffer and at the same time there is output event the focus will be stolen. We had similar issue in the past with `lsp-log` and there we went on implementing the following logic - if the user is in the end of the buffer - keep him in the end. At that time, I was unaware of the StackOverflow solution. What do you think?

Reply:
Ahh, I understand the problem with this implementation now. 

I will look into this stackoverflow solution and see if it works!

Thanks for the feedback! I really appreciate it.

Reply:
The solution from stackoverflow works as expected! The window point moves automatically if the point is at the end of the buffer, but does not move if the user moves the point somewhere else in the buffer.

Reply:
I missed that comment - please ignore the comments from the last review(except the one for redundant win). I will test the solution on my side later today and proceed with merging if everything is ok.

Reply:
OK sounds good!